File: components/events/FeedRefreshedEvent.ts
import { Environment, fetchQuery, graphql } from "relay-runtime"

import { createEventHook } from "components/EventsProvider"

export interface FeedRefreshedEvent {
  user_id: string
  feed_id: string
}

const fetchFeedQuery = graphql`
  query FeedRefreshedEventQuery($id: ID!) {
    node(id: $id) {
      id
      ... on Feed {
        title
        homePageURL
        micropubEndpoint
        refreshedAt
      }
    }
  }
`

export async function handleFeedRefreshed(
  environment: Environment,
  event: FeedRefreshedEvent
): Promise<void> {
  await fetchQuery(environment, fetchFeedQuery, { id: event.feed_id })
}

export const useFeedRefreshedEvent = createEventHook(
  "FeedRefreshed",
  handleFeedRefreshed
)

File: ios/Courier/Courier/Events/FeedRefreshedEvent.swift
import Combine
import Relay
import CourierGenerated

struct FeedRefreshedEvent: Decodable {
    var userID: String
    var feedID: String

    enum CodingKeys: String, CodingKey {
        case userID = "user_id"
        case feedID = "feed_id"
    }
}

private let query = graphql("""
query FeedRefreshedEventQuery($id: ID!) {
  feed(id: $id) {
    id
    title
    homePageURL
    micropubEndpoint
    refreshedAt
    refreshing
  }
}
""")

func handleFeedRefreshed(_ environment: Environment, _ event: FeedRefreshedEvent) -> AnyPublisher<Never, Error> {
    environment.fetchQuery(FeedRefreshedEventQuery(variables: .init(id: event.feedID)))
        .ignoreOutput()
        .eraseToAnyPublisher()
}
